Homework 9

Fwerfad opened the pull request.

Fwerfad committed:
homework3_v1

Fwerfad committed:
homework3_v1

Fwerfad committed:
homework3_v1

Fwerfad committed:
moved xml files to appropriate location

Fwerfad committed:
homework3_v2

Fwerfad committed:
homework3_v2

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v1

Fwerfad committed:
homework5_v2

Fwerfad committed:
homework7_init_commit

Fwerfad committed:
homework7_initial_commit

Fwerfad committed:
homework7_partly_done

Fwerfad committed:
homework7_v1

Fwerfad committed:
homework7_v2

Fwerfad committed:
homework9_v1

Fwerfad committed:
homework9_v1

Fwerfad committed:
homework9_v1

Fwerfad committed:
homework9_v1

Fwerfad committed:
homework9_v1

GrebNat commented on src/main/java/hw9/speller/dto/ResponseDto.java:
```diff
@@ -0,0 +1,16 @@
+@AllArgsConstructor
+@Accessors(chain = true)
+public class ResponseDto {
+    ArrayList<String> results;
```
что тут происходит? чя не понимаю, как можно такую дто использовать в контексте спеллера

Fwerfad replied:
Эта дтошка используется для хранения ответа от сервиса, сервис возвращает только текст, соответственно в ней только одно поле - список строк текста тегов s.
 Переименовал поле в texts.

GrebNat replied:
ответила там, где попытка дессериализации

GrebNat commented on src/main/java/hw9/speller/dto/TextDto.java:
```diff
@@ -0,0 +1,28 @@
+    public static int IGNORE_DIGITS = 2;
+    public static int IGNORE_URLS = 4;
+    public static int FIND_REPEAT_WORDS = 8;
+    public static int IGNORE_CAPITALIZATION = 512;
```
это константы, они не относятся к dto? т.к. просто являются  возможными значениями options. Вынеси их отдельно.
и второе - если это констнты, то должны бать final

Fwerfad replied:
Они относятся к возможным значениям options самого сервиса, вынес их в отдельный класс, поправил то, что они не final

GrebNat commented on src/main/java/hw9/speller/service/CommonService.java:
```diff
@@ -0,0 +1,87 @@
+import java.util.Properties;
+
+public class CommonService {
+    @SneakyThrows
```
для чтения пропертей создай отдельный класс

Fwerfad replied:
Сделал отдельный класс Properties

GrebNat commented on src/main/java/hw9/speller/service/CommonService.java:
```diff
@@ -0,0 +1,87 @@
+package hw9.speller.service;
```
в этом классе у тебя очень много дублированного кода. ОЧЕНЬ
пожалуйста, избавься от него любыми легальными способами

Fwerfad replied:
Сократил дублирование кода как мог

GrebNat commented on src/main/java/hw9/speller/service/RestTextAssertions.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public RestTextAssertions assertStatusCode(Response request, int expectedStatusCode) {
+        Assert.assertEquals(request.statusCode(), expectedStatusCode);
```
сообщение об ошибке

Fwerfad replied:
Добавил

GrebNat commented on src/main/java/hw9/speller/service/CommonService.java:
```diff
@@ -0,0 +1,87 @@
+        return props;
+    }
+
+    public Response getResponse(String text, int endpoint) {
```
эти методы не есть уровень Common Service.

тут ты описываешь как раз бизнес действия, по документации апи спеллера. Внимательно прочитай комент и исправь согласно тому, что я рассказывала на лекции последней и консультации.

GrebNat commented on src/main/java/hw9/speller/service/RestTextService.java:
```diff
@@ -0,0 +1,27 @@
+        return restTextService;
+    }
+
+    public ResponseDto getPrettyResult(Response request) {
```
Метод странен. Я просила для дессеризации использовать Gson? например. 
Зачем ты парвсишь ответ так сложно?

GrebNat replied:
Не исправлено. повторю, что для чтения результата запроса нужно использовать Gson  и дессериализовать его в DTO. Но не такое, как у тебя, только для тега s, а для всего ответа!!

Fwerfad replied:
Дополнил дтошку

Fwerfad replied:
Сделал дессериализацию через gson

GrebNat commented on src/main/java/hw9/speller/dto/TextDto.java:
```diff
@@ -0,0 +1,28 @@
+    public static int IGNORE_URLS = 4;
+    public static int FIND_REPEAT_WORDS = 8;
+    public static int IGNORE_CAPITALIZATION = 512;
+    private ArrayList<String> texts;
```
private ArrayList<String> texts;
    private String text;


не мешай обе dto в один класс. Если смущают дублирование полей (и это нормально) вспомни о НАСЛЕДОВАНИИ

GrebNat commented on src/test/java/hw9/speller/functional_level/CheckDifferentOptionsFuntionality.java:
```diff
@@ -0,0 +1,41 @@
+    @Test(dataProvider = "data-provider")
+    void checkResult(TextDto textDto) {
+        Response request = RestTextService.getInstance()
+                .getResponse(textDto.getText(), textDto.getEndpoint(), textDto.getOptions());
```
textDto.getEndpoint() - тесты с проверкой бизнес-действий не подразусевают передачу эндпоинта в тесте!!!
для этого и пишется отдельный метод с название ТИПА performCheckText(String text, String lang)....

GrebNat replied:
не исправлено

GrebNat replied:
ожидается тут действие тапиру RestTextService.getText("some_text")

Fwerfad replied:
Убрал эндпоинты из тестов в сам сервис

GrebNat commented on src/test/java/hw9/speller/functional_level/CheckDifferentOptionsFuntionality.java:
```diff
@@ -0,0 +1,41 @@
+                .getPrettyResult(request);
+        RestTextAssertions.getInstance()
+                .assertStatusCode(request, 200)
+                .assertTexts(textDto, actualResult);
```
assertStatusCode(request, 200) аналогично - не необходимости каждый тест перегружать проверкой еще и статус кода. Это все равно, что проверять, что wifi работает

GrebNat replied:
не исправлено

Fwerfad replied:
исправил

GrebNat commented on src/test/java/hw9/speller/functional_level/CheckGettingExpectedResult.java:
```diff
@@ -0,0 +1,37 @@
+package hw9.speller.functional_level;
```
аналогичные коменты

GrebNat commented on src/test/java/hw9/speller/functional_level/CheckGettingExpectedResults.java:
```diff
@@ -0,0 +1,45 @@
+package hw9.speller.functional_level;
```
аналогичные коменты.
нет смысла в подобных тестах для сдачи домашки

GrebNat commented on src/test/java/hw9/speller/low_level/CheckWrongFormatSubmitted.java:
```diff
@@ -0,0 +1,41 @@
+package hw9.speller.low_level;
```
аналогичные коменты

GrebNat commented on src/test/java/hw9/speller/low_level/CheckWrongLanguageSubmitted.java:
```diff
@@ -0,0 +1,38 @@
+package hw9.speller.low_level;
```
same

GrebNat commented on src/test/java/hw9/speller/low_level/YandexSpellerExampleTest.java:
```diff
@@ -0,0 +1,37 @@
+package hw9.speller.low_level;
```
просила не сдавать LOW Level тесты

Fwerfad replied:
Изначально это пример, из которого я раскуривал как работать с restassured. Удалил.

GrebNat commented on src/test/resources/hw9/test.properties:
```diff
@@ -0,0 +1,3 @@
+BaseUrl=https://speller.yandex.net/services/spellservice.json
+textEndpoint=/checkText
```
не всегда удачно оставлять эндпоинты в пропертях, но тут нормально

GrebNat reviewed (commented):
пересмотри лекции, нарушены некоторые моменты, на которых я специально останавливалась

Fwerfad committed:
Improved quality of code

Fwerfad committed:
Improved quality of code

Fwerfad committed:
Improved quality of code

Fwerfad committed:
Improved quality of code

Fwerfad committed:
fixed mvn clean test not working

GrebNat reviewed (commented):
ответила в комментариях

Fwerfad committed:
Moved all TextSpeller methods to RestTextService
Created noParams query at CommonService

Fwerfad committed:
Moved all TextSpeller methods to RestTextService
Created noParams query at CommonService
Added ResponseDto deserialization by gson

Fwerfad committed:
Moved all TextSpeller methods to RestTextService
Created noParams query at CommonService
Added ResponseDto deserialization by gson
